addons/hr/models/hr_plan.py

```diff
@@ -59,6 +82,142 @@ class HrPlan(models.Model):
     _name = 'hr.plan'
     _description = 'plan'
 
+    def _default_trigger_onboarding(self):
+        if self.plan_type == 'onboarding' and self.trigger:
+            return self.trigger
+        return 'manual'
+
+    def _default_trigger_offboarding(self):
+        if self.plan_type == 'offboarding' and self.trigger:
+            return self.trigger
+        return 'manual'
+
```

Comment:
Should be dropped. Use computed/stored/editabled fields instead, to make it work in all the cases.